lib/gc/ExecutionEngine/CPURuntime/MemoryPool.cpp

```diff
@@ -0,0 +1,277 @@
+//===-- MemoryPool.cpp - Runtime allocator memory pool ----------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <cassert>
+#include <iostream>
+#include <memory.h>
+#include <stdexcept>
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <sys/mman.h>
+#include <unistd.h>
+#endif
+
+#ifdef _MSC_VER
+#define __builtin_expect(EXP_, C) (EXP_)
+#endif
+
+#define likely(x) __builtin_expect(!!(x), 1)
+#define unlikely(x) __builtin_expect(!!(x), 0)
+namespace {
+// 4MB
+constexpr size_t threadlocalChunkSize = 4 * 1024 * 1024;
+// 16MB
+constexpr size_t mainChunkSize = 16 * 1024 * 1024;
+
+static constexpr size_t defaultAlignment = 64;
```

Comment:
Thanks for the suggestion. These values are fixed and have been chosen based on their proven performance on ICX, SPR, and EMR machines with our GC_V1 system.

We can certainly consider making these parameters configurable in the future, which would allow us to fine-tune performance for additional platforms as we expand our testing.